### DISCUSSION TITLE
Storybook Angular template interpolation for data attributes

### DISCUSSION THREAD

**Discussion by Sjeiti** [storybookjs/storybook#17930] [Help]
The documentation is unclear when it comes to passing data to data-attributes. 

Passing data to component inputs is done by regular Angular notation using brackets: `[inputname]="propertyname"`. For `ng-content` or anything outside the component one can use double curly braces: `{{propertyname}}`.

But setting a JSON string to a data attribute seems impossible.

Checkout below trial-and-error setup:

```
export default {
  title: 'Fuuuu'
}

const style = `<style>div[data-a] {
    font-family: Courier, monospace;
    white-space: pre;
    font-weight: 600;
}div[data-a]:after {
    content: ${'abcde'.split('').map(s=>`'\\A ${s}: 'attr(data-${s})`).join('')+';'}
    font-weight: 400;
}</style>`

export const Fuuuu = () => ({
  template: `${style}<div
  data-a="a"
  data-b="baz"
  data-c="{{baz}}"
  [data-d]="baz"
  [dataE]="baz"
>Fuuuu {{baz}}</div>`,
  props: {
    baz: 'qux'
  }
})

//

const toString = obj => JSON.stringify(obj)
    .replace(/"/g, '') // <-- this does not fail: comment it and uncomment below
    // .replace(/"/g, '\\"')
    // .replace(/{/g, '{{\'{\'}}')

export const Aargh = Object.assign((args) => ({
  template: `
${style}<div
  data-a="${args.baz}"
  data-b="${toString(args.abc)}"
>Fuuuu {{baz}} ${args.baz}</div>`,
  props: {
    baz: 'qux'
  }
}), {
  args: {
    baz: 'bar',
    abc: {def: 123, ghi: ['a', 'b']}
  }
})
```

One would expect page 'Fuuuu' to contain a value for `[data-c]`.

In the page 'Aargh' the value is passed by `props` but this will cause an error (unterminated tag).

any help is greatly appreciated,

Ron

**Comment by zergeborg** [comment]
I've also come across this issue today. And I am scratching my head.

I would add that in my case not only the template interpolation is problematic, but more advanced stuff like stamping out templates and/or components with `*ngTemplateOutlet` and `*ngComponentOutlet` do not work.

How can someone pass data to the templates? And how can one pass other template/component instances to the reusable components that story is about???

Also, why are all the examples in the doc use a template string??? That's an anti pattern according to the Angular style guide.
https://angular.io/guide/styleguide#extract-templates-and-styles-to-their-own-files

The documentation is absolutely lacking. SMH.
